src/components/tl/ucp/allgather/allgather_ring.c

```diff
@@ -67,6 +69,8 @@ void ucc_tl_ucp_allgather_ring_progress(ucc_coll_task_t *coll_task)
         if (UCC_INPROGRESS == ucc_tl_ucp_test(task)) {
             return;
         }
+        step = use_loopback ? task->tagged.send_posted - 1
```

Comment:
Should this line check for inplace as well? If use_loopback is true and inplace is false, will step be incorrect?

Reply:
Similar for neighbor exchange